### PR TITLE
Add a getter and some unit tests for XYPlot.autorange()

### LINKED ISSUE #2176: XYPlot.autorange() should function as a getter

### PR DESCRIPTION
**Pull request by dbt**
Fixes #2176.

### CODE REVIEW

File: src/components/plots/xyPlot.ts [hunk 1]
```diff
@@ -156,10 +160,19 @@ module Plottable {
      *   "x" will adjust the x Scale in relation to changes in the y domain.
      *   "y" will adjust the y Scale in relation to changes in the x domain.
      *   "none" means neither Scale will change automatically.
-     *
      * @returns {XYPlot} The calling XYPlot.
      */
-    public autorange(scaleName: string) {
+    public autorange(scaleName: string): XYPlot<X, Y>;
+    public autorange(scaleName?: string): any {
+      if (typeof(scaleName) === "undefined") {
```
Comment (aicioara) [inline src/components/plots/xyPlot.ts]: The convention around the codebase is 
```typescript
if (scaleName == null)
```

We discussed we will change to the bellow soon, but for now, the above is consistent 
```typescript
if (scaleName === void(0))
```

File: test/components/plots/xyPlotTests.ts [hunk 2]
```diff
@@ -27,6 +27,23 @@ describe("Plots", () => {
           .renderTo(svg);
     });
 
+    it("autorange getters", () => {
```
Comment (aicioara) [inline test/components/plots/xyPlotTests.ts]: `autorange() getter`

File: test/components/plots/xyPlotTests.ts [hunk 3]
```diff
@@ -27,6 +27,23 @@ describe("Plots", () => {
...
+      plot.autorange("x");
```
Comment (aicioara) [inline test/components/plots/xyPlotTests.ts]: Maybe test the default as well (I believe it should be "none"?). i.e. assert `plot.autorange()` before setting it explicitly

**Comment by aicioara** [comment]
Minor style issues

**Comment by aicioara** [comment]
I restarted the builds. Looks like Travis was having an issue yesterday with `nvm` version

File: test/components/plots/xyPlotTests.ts [hunk 4]
```diff
@@ -27,6 +27,24 @@ describe("Plots", () => {
...
     });
 
+    it("autorange() getter", () => {
+      assert.deepEqual(plot.autorange(), "none");
+      assert.deepEqual(plot.autorange("x"), plot);
+      assert.deepEqual(plot.autorange(), "x");
+      plot.autorange("y");
+      assert.deepEqual(plot.autorange(), "y");
+      plot.autorange("none");
+      assert.deepEqual(plot.autorange(), "none");
```
Comment (jtlan) [inline test/components/plots/xyPlotTests.ts]: Should be `assert.strictEqual` here.

Also we should have a message on the asserts, so if one of them fails in the test it's easy to find which one. For example:
```Typescript
assert.strictEqual(plot.autorange("x"), plot, "autorange(arg) returns the calling Plot");
```

**Comment by jtlan** [comment]
[NOQE] since the tests basically cover the entire case, but [-1] for some test structure.

**Comment by jtlan** [comment]
[+1] GIM

**Pull request merged**